Mathlib/Topology/UniformSpace/DiscreteUniformity.lean

```diff
@@ -0,0 +1,102 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir, María Inés de Frutos Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johannes Hölzl, Mario Carneiro, Antoine Chambert-Loir, María Inés de Frutos Fernández
+-/
+import Mathlib.Topology.UniformSpace.Cauchy
+import Mathlib.Topology.Algebra.UniformGroup.Defs
+
+/-! # Discrete uniformity
+
+The discrete uniformity is the smallest possibly uniformity, the one for which
+the diagonal is an entourage of itself.
+
+It induces the discrete topology.
+
+It is complete.
+
+-/
+
+open Filter UniformSpace
+
+/-- The discrete uniformity -/
+class DiscreteUniformity (X : Type*) [u : UniformSpace X] : Prop where
+  eq_bot: u = ⊥
+
+namespace DiscreteUniformity
+
+/-- The bot uniformity is the discrete uniformity -/
+instance (X : Type*) : @DiscreteUniformity X ⊥ :=
+  @DiscreteUniformity.mk X ⊥ rfl
+
+variable (X : Type*) [u : UniformSpace X] [DiscreteUniformity X]
+
+theorem eq_principal_idRel : uniformity X = principal idRel :=
+  le_antisymm
+    (by simpa [uniformSpace_eq_bot, ← le_principal_iff] using eq_bot)
+    refl_le_uniformity
+
+theorem iff_eq_principal_idRel {X : Type*} [UniformSpace X] :
+    DiscreteUniformity X ↔ uniformity X = principal idRel :=
+  ⟨fun _ ↦ eq_principal_idRel X, fun h ↦ { eq_bot := by ext; rw [h]; rfl }⟩
+
+/-- The discrete uniformity induces the discrete topology -/
+instance : DiscreteTopology X where
+  eq_bot := by
+    convert UniformSpace.toTopologicalSpace_bot
+    exact eq_bot
+
+theorem idRel_mem_uniformity : idRel ∈ uniformity X := by
+  simp only [eq_principal_idRel, mem_principal, subset_refl]
```

Comment:
Please add the `iff` version as well, probably using [UniformSpace.uniformSpace_eq_bot](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/UniformSpace/Basic.html#UniformSpace.uniformSpace_eq_bot).